actions/class.ItemContent.php

```diff
@@ -156,16 +142,16 @@ public function upload()
             } else {
                 $valid = false;
                 // OR the extension is part of the filter and it correspond to the mime type
-                foreach ($filters as $filter) {
+                foreach ($params['filters'] as $filter) {
                     if (
                         $filter['mime'] === $mime &&
-                        (!isset($filter['extension']) || $filter['extension'] === \tao_helpers_File::getFileExtention($fileTmpName))
+                        (!isset($filter['extension']) || $filter['extension'] === tao_helpers_File::getFileExtention($fileTmpName))
```

Comment:
Maybe we can store `tao_helpers_File::getFileExtension($fileTmpName)` in a separate variable and call it before `foreach`?

```
$fileExtension = tao_helpers_File::getFileExtention($fileTmpName);

foreach ($params['filters'] as $filter) {
    if (
        $filter['mime'] === $mime
        && (!isset($filter['extension']) || $filter['extension'] === $fileExtension)
    ) {
        $valid = true;
    }
}
```

Reply:
Done